### PR TITLE
Accept non-standard bools in more CUDA kernels

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78957

This fixes all remaining CUDA kernels, except those using `cub` or
`thrust`, to accept boolean tensors with values other than 1 or 0.

I do this by using `c10::load` in more places, and also adding a
`load_vector` helper into `MemoryAccess.cuh` that does the same thing
for vectorized loads.

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -11220,7 +11217,8 @@ def error_inputs_mean(op_info, device, **kwargs):
            supports_forward_ad=True,
            supports_fwgrad_bwgrad=True,
            skips=(
-               DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_non_standard_bool_values'),
+               DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_non_standard_bool_values',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:11220]: Nice

**Comment by mruberry** [comment]
I added the ciflow/trunk label to trigger more tests, just in case, since this is modifying the OpInfos

**Comment by peterbell10** [comment]
@mruberry PTAL

**Review by mruberry — APPROVED** [review]
Thanks @peterbell10!

**Pull request closed without merging**